src/frontend/Account.ts

```diff
@@ -235,59 +220,31 @@ class Account {
     return this.magic?.user ? await this.magic.user.isLoggedIn() : false;
   }
 
-  /**
-   * Get network params (to be passed to the `wallet_addEthereumChain` rpc method) for the given chain id
-   */
-  private _getNetworkParams() {
-    const testnetParams = {
-      chainId: "0x13881",
-      chainName: "Mumbai Testnet",
-      nativeCurrency: {
-        name: "Matic",
-        symbol: "MATIC",
-        decimals: 18,
-      },
-      rpcUrls: ["https://matic-mumbai.chainstacklabs.com"],
-      blockExplorerUrls: ["https://mumbai.polygonscan.com/"],
-    };
-
-    const mainnetParams = {
-      chainId: "0x89",
-      chainName: "Polygon Mainnet",
-      nativeCurrency: {
-        name: "Matic",
-        symbol: "MATIC",
-        decimals: 18,
-      },
-      rpcUrls: ["https://polygon-rpc.com"],
-      blockExplorerUrls: ["https://polygonscan.com"],
-    };
-
-    const params = CHAIN_ID === "0x89" ? mainnetParams : testnetParams;
-    return params;
+  async getChainId(): Promise<string | undefined> {
+    if (!this.walletProvider) return;
+    return await this.walletProvider.getNetwork().then(network => network.chainId.toString());
```

Comment:
The `await` is not needed.
Also I think this fits the style of the codebase more: 

```typescript
const {chainId} = await this.walletProvider.getNetwork();
return chainId.toString();
```